hw/bsp/hifive1/src/hal_bsp.c

```diff
@@ -109,47 +84,24 @@ hal_bsp_init(void)
     int rc;
 
     (void)rc;
-#if MYNEWT_VAL(TIMER_0)
-    rc = hal_timer_init(0, (void *)&fe310_pwm2);
-    assert(rc == 0);
-#endif
-#if MYNEWT_VAL(TIMER_1)
-    rc = hal_timer_init(1, (void *)&fe310_pwm1);
-    assert(rc == 0);
-#endif
-#if MYNEWT_VAL(TIMER_2)
-    rc = hal_timer_init(2, (void *)&fe310_pwm0);
-    assert(rc == 0);
-#endif
-
-#if (MYNEWT_VAL(OS_CPUTIME_TIMER_NUM) >= 0)
-    rc = os_cputime_init(MYNEWT_VAL(OS_CPUTIME_FREQ));
-    assert(rc == 0);
-#endif
+    fe310_periph_create();
 
-#if MYNEWT_VAL(SPI_1)
+#if MYNEWT_VAL(SPIFLASH)
 #if MYNEWT_VAL(BUS_DRIVER_PRESENT)
-    rc = bus_spi_dev_create("spi1", &spi1_bus, (struct bus_spi_dev_cfg *)&spi1_cfg);
-    assert(rc == 0);
-#else
-    rc = hal_spi_init(1, NULL, HAL_SPI_TYPE_MASTER);
-    assert(rc == 0);
-#endif
-#endif
-#if MYNEWT_VAL(SPI_2)
-#if MYNEWT_VAL(BUS_DRIVER_PRESENT)
-    rc = bus_spi_dev_create("spi2", &spi2_bus, (struct bus_spi_dev_cfg *)&spi2_cfg);
-    assert(rc == 0);
-#else
-    rc = hal_spi_init(2, NULL, HAL_SPI_TYPE_MASTER);
+    /* Create external flash dev */
+    rc = spiflash_create_spi_dev(&spiflash_dev.dev,
+        MYNEWT_VAL(BSP_FLASH_SPI_NAME), &flash_spi_cfg);
+
     assert(rc == 0);
 #endif
 #endif
+}
 
-#if MYNEWT_VAL(UART_0)
-    rc = os_dev_create((struct os_dev *) &os_bsp_uart0, "uart0",
-      OS_DEV_INIT_PRIMARY, 0, uart_hal_init, (void *)&os_bsp_uart0_cfg);
-    assert(rc == 0);
-#endif
+int
+hal_bsp_hw_id(uint8_t *id, int max_len)
+{
+    /* XXX figure out what to put here */
+    memset(id, 0x00, max_len);
 
+    return max_len;
```

Comment:
Maybe you should just return `0` and do nothing?

Reply:
You are right, fixed